lib/Parser/Parse.cpp

```diff
@@ -10925,7 +10925,8 @@ ParseNodePtr Parser::Parse(LPCUTF8 pszSrc, size_t offset, size_t length, charcou
         }
     }
 
-    if (isModuleSource && !isDeferred)
+    // It's possible for the module global to be defer-parsed in debug scenarios.
+    if (isModuleSource && (!isDeferred || (isDeferred && grfscr & fscrGlobalCode)))
```

Comment:
Is it necessary that we allow the module global be deferred? (I assume this logic is just to make a synthetic scenario happy and that we would never do this in reality?)